### PR TITLE
Ammo casings and speedloader sprite changes

### PR DESCRIPTION
**Pull request by comma**
Moved them into single icon dmis
Changed the way they're generated completely.
Casings now have base color, color of bullet, and optional marking color, and icon is composed of those. Added some rudimentary inventory icons too.
Speedloaders now do some arcane juju to dynamically assemble an inventory icon based on every casing's colors. In-world speedloader icons were added, with way less complicated icongen.
image
image

### CODE REVIEW

File: code/modules/projectiles/ammunition/boxes.dm [hunk 1]
```diff
@@ -1,10 +1,47 @@
 /obj/item/ammo_magazine/speedloader
-	icon_state = "spdloader_magnum"
+	icon_state = "world"
 	caliber = CALIBER_PISTOL_MAGNUM
 	ammo_type = /obj/item/ammo_casing/pistol/magnum
 	material = MAT_STEEL
 	max_ammo = 6
-	multiple_sprites = 1
+	on_mob_icon = 'icons/obj/ammo/speedloader.dmi'
+	var/list/global/bullet_offsets = list(
```
Comment (Lohikar) [inline code/modules/projectiles/ammunition/boxes.dm:8]: Please no implicit strings.

File: code/modules/projectiles/ammunition/boxes.dm [hunk 2]
```diff
@@ -1,10 +1,47 @@
...
+	)
+
+/obj/item/ammo_magazine/speedloader/on_update_icon()
+	cut_overlays()
+	if(length(stored_ammo))
+		var/list/overlays_to_add = list()
+		switch(icon_state)
+			if("world")
+				var/ammo_state = "world-some"
+				if(length(stored_ammo) == max_ammo)
```
Comment (Lohikar) [inline code/modules/projectiles/ammunition/boxes.dm]: might be more concise to use a switch here?

File: code/modules/projectiles/ammunition/boxes.dm [hunk 3]
```diff
@@ -1,10 +1,47 @@
...
+		list(x = -2, y = -7),
+		list(x = 0, y = -10),
+		list(x = 2, y = -7),
+		list(x = 2, y = -3)
+	)
+
+/obj/item/ammo_magazine/speedloader/on_update_icon()
+	cut_overlays()
+	if(length(stored_ammo))
+		var/list/overlays_to_add = list()
```
Comment (Lohikar) [inline code/modules/projectiles/ammunition/boxes.dm]: There's not much point using an overlays list with `add_overlay()`.

File: code/modules/projectiles/ammunition/boxes.dm (hunk 1 above)
Comment (comma) [reply code/modules/projectiles/ammunition/boxes.dm:8]: fixed

File: code/modules/projectiles/ammunition/boxes.dm (hunk 2 above)
Comment (comma) [reply code/modules/projectiles/ammunition/boxes.dm]: can't, max_ammo is not constant so it rees

File: code/modules/projectiles/ammunition/boxes.dm (hunk 3 above)
Comment (comma) [reply code/modules/projectiles/ammunition/boxes.dm]: fixed

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**